### ISSUE TITLE
DMA ID not in some DMA summary sheets

### ISSUE THREAD

**Issue by cooperjaXC** [ccaoa/google-ece-trends#11]
When old versions of the summary statistics for DMA data do not already have the DMA ID field (`dma_id`), the code errors out when it attempts to retain only the important, specified columns (seen in the command below)

https://github.com/ccaoa/google-ece-trends/blob/ca621879866ae75e52709cc0572ae26e54916fd3/summarize.py#L326-L327

**The solution is to add the DMA IDs to the summary DF while the sumstats are calculated and before the filtering down of columns in the DF occurs.**

---

This issue arose from Alex's personal computer's run of the GTrends pull. Below is the copy of the Texas DMA summary stats' error.
```
Starting Google Trends data pull.
Traceback (most recent call last):
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\late2022_datapulls.py", line 326, in <module>
    full_run_gtrends()
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\late2022_datapulls.py", line 318, in full_run_gtrends
    agg.summarize_collected_data(targ_sumfiles_listdir, suppress_prints=False)
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\summarize.py", line 402, in summarize_collected_data
    calc_sumstats(sd)
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\summarize.py", line 327, in calc_sumstats
    sumstats_df = sumstats_df[sumstatvars]
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\frame.py", line 3464, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1)[1]
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\indexing.py", line 1314, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis)
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\indexing.py", line 1377, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: "['dma_id'] not in index"
```
Further debugging led to realizing which file was the problem. Below shows the successful OR DMA summarize run because it already had the DMA ID column. Texas fails because it did not. 
```
DMA FILE!: or_dma_20200214-20210214.xlsx
  'dma_id' in DF after insert command? True
10 / 24 processed:    or_dma_20200214-20210214.xlsx
11 / 24 processed:    or_time_20200214-20210214.xlsx
DMA FILE!: tx_dma_20210321-20210421.xlsx
  'dma_id' in DF after insert command? False
Traceback (most recent call last):
  File "C:/Users/acc-s/Documents/Coding/Git/GitHub/ccaoa_github/google-ece-trends/summarize.py", line 580, in <module>
    summarize_all_summary_data(sum_data_pth)
  File "C:/Users/acc-s/Documents/Coding/Git/GitHub/ccaoa_github/google-ece-trends/summarize.py", line 430, in summarize_all_summary_data
    summarize_collected_data(all_agg_files, suppress_prints=suppress_prints)
  File "C:/Users/acc-s/Documents/Coding/Git/GitHub/ccaoa_github/google-ece-trends/summarize.py", line 404, in summarize_collected_data
    calc_sumstats(sd)
  File "C:/Users/acc-s/Documents/Coding/Git/GitHub/ccaoa_github/google-ece-trends/summarize.py", line 329, in calc_sumstats
    sumstats_df = sumstats_df[sumstatvars]
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\frame.py", line 3464, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1)[1]
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\indexing.py", line 1314, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis)
  File "C:\Users\acc-s\Documents\Coding\Git\GitHub\ccaoa_github\google-ece-trends\py_venv\venv\lib\site-packages\pandas\core\indexing.py", line 1377, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: "['dma_id'] not in index"

Process finished with exit code 1
```

**Comment by cooperjaXC** [comment]
It may be good to completely reassign the DMA IDs to the summary sheet every time it is calculated because mismatching errors between name and ID were discovered as a part of bugfixing this issue. Reassigning with every summary run guarantees that there won't be mismatching name~ID errors.

**Comment by cooperjaXC** [comment]
Closed by pull request #12
